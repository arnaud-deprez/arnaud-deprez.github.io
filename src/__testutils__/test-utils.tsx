import React from 'react'
import { render } from '@testing-library/react'
import { LocationProvider, createHistory } from '@gatsbyjs/reach-router'

const history = createHistory(window)

const AllTheProviders = ({ children }) => {
  return <LocationProvider history={history}>{children}</LocationProvider>
}

const customRender = (ui, options = {}) => render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
