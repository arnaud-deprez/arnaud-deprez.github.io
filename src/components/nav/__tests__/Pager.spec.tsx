import React from 'react'
import { render } from '../../../__testutils__/test-utils'
import { Pager } from '..'

describe('<Pager />', () => {
  it('should not render if there is only one page', () => {
    const { asFragment } = render(<Pager prefix="/test" page={1} total={1} />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('should not render previous and first button on page 1', () => {
    const { asFragment } = render(<Pager prefix="/test" page={1} total={10} />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('should not render next and last button on last page', () => {
    const { asFragment } = render(<Pager prefix="/test" page={10} total={10} />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('should render 5 page items with first, previous, next and last button', () => {
    const { asFragment } = render(<Pager prefix="/test" page={5} total={10} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
