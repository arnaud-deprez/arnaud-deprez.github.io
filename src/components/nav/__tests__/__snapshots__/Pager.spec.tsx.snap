// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pager /> should not render if there is only one page 1`] = `
<DocumentFragment>
  <ul
    class="pagination"
  />
</DocumentFragment>
`;

exports[`<Pager /> should not render next and last button on last page 1`] = `
<DocumentFragment>
  <ul
    class="pagination"
  >
    <li
      class="page-item"
    >
      <a
        as="[object Object]"
        class="page-link"
        href="/test"
      >
        <span
          aria-hidden="true"
        >
          «
        </span>
        <span
          class="visually-hidden"
        >
          First
        </span>
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        as="[object Object]"
        class="page-link"
        href="/test/9"
      >
        <span
          aria-hidden="true"
        >
          ‹
        </span>
        <span
          class="visually-hidden"
        >
          Previous
        </span>
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        as="[object Object]"
        class="page-link"
        href="/test/6"
      >
        6
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        as="[object Object]"
        class="page-link"
        href="/test/7"
      >
        7
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        as="[object Object]"
        class="page-link"
        href="/test/8"
      >
        8
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        as="[object Object]"
        class="page-link"
        href="/test/9"
      >
        9
      </a>
    </li>
    <li
      class="page-item active"
    >
      <span
        class="page-link"
      >
        10
        <span
          class="visually-hidden"
        >
          (current)
        </span>
      </span>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`<Pager /> should not render previous and first button on page 1 1`] = `
<DocumentFragment>
  <ul
    class="pagination"
  >
    <li
      class="page-item active"
    >
      <span
        class="page-link"
      >
        1
        <span
          class="visually-hidden"
        >
          (current)
        </span>
      </span>
    </li>
    <li
      class="page-item"
    >
      <a
        as="[object Object]"
        class="page-link"
        href="/test/2"
      >
        2
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        as="[object Object]"
        class="page-link"
        href="/test/3"
      >
        3
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        as="[object Object]"
        class="page-link"
        href="/test/4"
      >
        4
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        as="[object Object]"
        class="page-link"
        href="/test/5"
      >
        5
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        as="[object Object]"
        class="page-link"
        href="/test"
      >
        <span
          aria-hidden="true"
        >
          ›
        </span>
        <span
          class="visually-hidden"
        >
          Next
        </span>
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        as="[object Object]"
        class="page-link"
        href="/test/10"
      >
        <span
          aria-hidden="true"
        >
          »
        </span>
        <span
          class="visually-hidden"
        >
          Last
        </span>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`<Pager /> should render 5 page items with first, previous, next and last button 1`] = `
<DocumentFragment>
  <ul
    class="pagination"
  >
    <li
      class="page-item"
    >
      <a
        as="[object Object]"
        class="page-link"
        href="/test"
      >
        <span
          aria-hidden="true"
        >
          «
        </span>
        <span
          class="visually-hidden"
        >
          First
        </span>
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        as="[object Object]"
        class="page-link"
        href="/test/4"
      >
        <span
          aria-hidden="true"
        >
          ‹
        </span>
        <span
          class="visually-hidden"
        >
          Previous
        </span>
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        as="[object Object]"
        class="page-link"
        href="/test/3"
      >
        3
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        as="[object Object]"
        class="page-link"
        href="/test/4"
      >
        4
      </a>
    </li>
    <li
      class="page-item active"
    >
      <span
        class="page-link"
      >
        5
        <span
          class="visually-hidden"
        >
          (current)
        </span>
      </span>
    </li>
    <li
      class="page-item"
    >
      <a
        as="[object Object]"
        class="page-link"
        href="/test/6"
      >
        6
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        as="[object Object]"
        class="page-link"
        href="/test/7"
      >
        7
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        as="[object Object]"
        class="page-link"
        href="/test/4"
      >
        <span
          aria-hidden="true"
        >
          ›
        </span>
        <span
          class="visually-hidden"
        >
          Next
        </span>
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        as="[object Object]"
        class="page-link"
        href="/test/10"
      >
        <span
          aria-hidden="true"
        >
          »
        </span>
        <span
          class="visually-hidden"
        >
          Last
        </span>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
